node/network/bridge/src/lib.rs

```diff
@@ -570,31 +590,60 @@ async fn run_network<N: Network>(
 			Action::Abort => return Ok(()),
 
 			Action::SendValidationMessage(peers, msg) => send_message(
-					&mut net,
+					&mut net.network_service,
 					peers,
 					PeerSet::Validation,
 					WireMessage::ProtocolMessage(msg),
 			).await?,
 
 			Action::SendCollationMessage(peers, msg) => send_message(
-					&mut net,
+					&mut net.network_service,
 					peers,
 					PeerSet::Collation,
 					WireMessage::ProtocolMessage(msg),
 			).await?,
 
-			Action::ConnectToValidators(_peer_set, _validators, _res) => {
-				// TODO: https://github.com/paritytech/polkadot/issues/1461
-			}
+			Action::ConnectToAuthorities(peer_set, authorities, res) => {
+				let priority_group = match peer_set {
+					PeerSet::Validation => "parachain_validation".to_owned(),
+					PeerSet::Collation => "parachain_collation".to_owned(),
+				};
+
+				// collect multiaddress of authorities
+				let mut multiaddresses = HashSet::new();
+				for authority in authorities.iter().cloned() {
+					let result = net.authority_discovery_service.get_addresses_by_authority_id(authority).await;
+					if let Some(addresses) = result {
+						// we might have several `PeerId`s per `AuthorityId` depending on the number of sentry nodes,
+						// so in theory we might want to limit max number of sentries per node to connect to,
+						// in practice the number of sentries per node is low and they are going to be removed soon
+						// https://github.com/paritytech/substrate/issues/6845
+						for addr in addresses {
+							multiaddresses.insert(addr);
+						}
```

Comment:
I don't understand this loop. The addresses returned by `get_addresses_by_authority_id` are also deduplicated.

Reply:
The loop is not about deduplication, it just collects all multiaddresses. The reason it uses a `HashMap` and not a `Vec` is because `set_priority_group` accepts a `HashSet`. But we don't want to call `set_priority_group` https://github.com/paritytech/polkadot/pull/1699#discussion_r486957653, so it'll be changed.